src/frontend/src/expr/type_inference/func.rs

```diff
@@ -451,11 +417,9 @@ fn infer_type_for_special(
                     // when neither type is available, default to `varchar[]`
                     // when one side is unknown and other side is list, use that list type
                     let t = t.unwrap_or_else(|| DataType::List(DataType::Varchar.into()));
-                    let inputs_owned = std::mem::take(inputs);
-                    *inputs = inputs_owned
-                        .into_iter()
-                        .map(|e| e.cast_implicit(t.clone()))
-                        .try_collect()?;
+                    for input in &mut *inputs {
```

Comment:
This was suggested by the clippy. 🥲
```
warning: it is more concise to loop over references to containers instead of using explicit iteration methods
   --> src/frontend/src/expr/type_inference/func.rs:420:34
    |
420 |                     for input in inputs.iter_mut() {
    |                                  ^^^^^^^^^^^^^^^^^ help: to write this more concisely, try: `&mut *inputs`
```